src/js/payments-methods/express/_payment-methods.js

```diff
@@ -45,22 +45,32 @@ const AmazonPayInfo = ( props ) => {
     const { amazonBilling, amazonShipping } = settings.amazonAddress;
 
     useEffect( () => {
-        const unsubscribe = props.eventRegistration.onCheckoutValidationBeforeProcessing(
+        const unsubscribe = props.eventRegistration.onCheckoutValidation(
             async () => {
                 for ( const shippingField in amazonShipping ) {
-                    if ( amazonShipping[ shippingField ] !== shippingAddress[ shippingField ] ) {
-                        return {
-                            errorMessage: __( 'We were expecting "', 'woocommerce-gateway-amazon-payments-advanced' ) + amazonShipping[ shippingField ] + __( '" but we received "', 'woocommerce-gateway-amazon-payments-advanced' ) + shippingAddress[ shippingField ] + __( '" instead for the Shipping field "', 'woocommerce-gateway-amazon-payments-advanced' ) + getCheckOutFieldsLabel( shippingField, 'shipping' ) + __( '". Please make any changes to your Shipping details through Amazon.', 'woocommerce-gateway-amazon-payments-advanced' )
-                        };
+                     // Values are the same as expected. Bail.
+                    if (amazonShipping[ shippingField ] === shippingAddress[ shippingField ]) {
+                        continue;
                     }
+            
+                    const checkoutFieldLabel = getCheckOutFieldsLabel( shippingField, 'shipping' );
+                    // Field not present in the form, as a result value can't be supplied. Bail.
+                    if ( false === checkoutFieldLabel ) {
+                        continue;
+                    }
+            
+                    // Field present in the form but value mismatch. Return error.
+                    return {
+                        errorMessage: __( 'We were expecting "', 'woocommerce-gateway-amazon-payments-advanced' ) + amazonShipping[ shippingField ] + __( '" but we received "', 'woocommerce-gateway-amazon-payments-advanced' ) + shippingAddress[ shippingField ] + __( '" instead for the Shipping field "', 'woocommerce-gateway-amazon-payments-advanced' ) + checkoutFieldLabel + __( '". Please make any changes to your Shipping details through Amazon.', 'woocommerce-gateway-amazon-payments-advanced' )
```

Comment:
Do we have JS functions available here for sprintf-alike formatting? Then it'd be just one new string fo translations.

Reply:
Your comment makes a lot of sense. Updating. Just wanted to mention that the string parts have been released already though so we might end up breaking already created translations if there are any.

Reply:
Acknowledged. Worth checking translations and asking to update those if we have a solution for my suggestion.
In the long run that simplifies translation maintenance.

Reply:
Updated to use sprintf. Checked the translations and only in German there are translations for those string awaiting approval. https://translate.wordpress.org/projects/wp-plugins/woocommerce-gateway-amazon-payments-advanced/stable/de/default/?filters%5Bstatus%5D=waiting